### PR TITLE
feat: Include 2D Meshing workflow.

### PR DESCRIPTION
**Pull request by prmukherj**
This exposes "2D Meshing" and "Topology Based Meshing" as part of the enhanced meshing workflow.

Please note:
This is being released to be integrated with PyConsole. There are a lot of restructuring opportunities as well as need to document and add tests for these new workflows. These will be done as part of other PRs.

### CODE REVIEW

**Comment by prmukherj** [comment]
Did some high level testing from PyConsole as well. Seems al right.

**Comment by seanpearsonuk** [comment]
Are these both beta in Fluent meshing workflows?

**Comment by seanpearsonuk** [comment]
> > Are these both beta in Fluent meshing workflows @2024R2?
> 
> Not the 2D Meshing.

I'm running the 2024R2 launcher. It only shows 3D option there, but it stays like that even with beta enabled so I wasn't sure what to make of it.

File: src/ansys/fluent/core/session_pure_meshing.py [hunk 1]
```diff
@@ -107,6 +107,24 @@ def fault_tolerant(self):
...
+    def topology_based(self):
+        """Get a new topology based meshing workflow.
+
+        Raises
+        ------
+        RuntimeError
+            If beta features are not enabled in Fluent.
+        """
+        if not self.scheme_eval.scheme_eval("(is-beta-feature-available?)"):
+            raise RuntimeError("Topology Based Meshing is a fluent beta feature.")
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_pure_meshing.py:124]:
```suggestion
            raise RuntimeError("Topology-based Meshing is a beta feature in Fluent.")
```

File: src/ansys/fluent/core/session_pure_meshing.py [hunk 2]
```diff
@@ -107,6 +107,24 @@ def fault_tolerant(self):
         self._base_meshing.fault_tolerant_workflow.reinitialize()
         return self._base_meshing.fault_tolerant_workflow
 
+    def two_dimensional_meshing(self):
+        """Get a new 2D meshing workflow."""
+        self._base_meshing.two_dimensional_meshing_workflow.reinitialize()
+        return self._base_meshing.two_dimensional_meshing_workflow
+
+    def topology_based(self):
+        """Get a new topology based meshing workflow.
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_pure_meshing.py:116]:
```suggestion
        """Get a new topology-based meshing workflow.
```

**Comment by prmukherj** [comment]
> > > Are these both beta in Fluent meshing workflows @2024R2?
> > 
> > 
> > Not the 2D Meshing.
> 
> I'm running the 2024R2 launcher. It only shows 3D option there, but it stays like that even with beta enabled so I wasn't sure what to make of it.

You can find the option in the meshing workflow drop-down.

File: src/ansys/fluent/core/meshing/meshing_workflow.py [hunk 3]
```diff
@@ -146,3 +146,67 @@ def pm_file_management(self):
...
+            and not self._meshing.GlobalSettings.EnablePrime2dMeshing()
+        ):
+            raise RuntimeError(
+                "'2D Meshing' objects are inaccessible from other workflows."
+            )
+        return super().__getattribute__(item)
+
+
+class TopologyBasedMeshingWorkflow(MeshingWorkflow):
+    """Provides topology based meshing specialization of the workflow wrapper."""
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply src/ansys/fluent/core/meshing/meshing_workflow.py:184]: will check it.

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Pull request merged**